playground/src/utils/helper_functions.tsx

```diff
@@ -55,8 +59,101 @@ export function checkField(jsonField: JSON, key: string): [boolean, string] {
     const errorMessage = `${key} must contain at least one number, 
 and it can only contain numbers. \n\n`;
 
-    return [false, errorMessage];
+    return { isValid: false, errorMessage: errorMessage };
+  }
+
+  return { isValid: true, errorMessage: "" };
+}
+
+/**
+ * We check the inputs and return true or false. We allow:
+ * - an empty input, and
+ * - a valid JSON object containing stack_init or advice_tape (or both)
+ * if the values are numbers
+ */
+export function checkInputs(jsonString: string): checkedData {
+  if (jsonString === "") {
+    return { isValid: true, errorMessage: "" };
+  }
+
+  let jsonInput!: JSON;
+
+  try {
+    jsonInput = JSON.parse(jsonString);
+  } catch (e: any) {// eslint-disable-line @typescript-eslint/no-explicit-any
+
+    const errorMessage = `Miden VM Inputs need to be a valid JSON object:
+${e.message}`;
+    return { isValid: false, errorMessage: errorMessage };
+  }
+
+  if (
+    !Object.keys(jsonInput).includes("stack_init") &&
+    !Object.keys(jsonInput).includes("advice_tape")
+  ) {
+    const errorMessage = `Miden VM Inputs can be empty or 
+we need either a stack_init or 
+an advice_tape.`;
+
+    return { isValid: false, errorMessage: errorMessage };
+  }
+
+  if (!checkFields(jsonInput).isValid) {
+    const errorMessage = checkFields(jsonInput).errorMessage;
+
+    return { isValid: false, errorMessage: errorMessage };
+  }
```

Comment:
In general, I would avoid calling the same function multiple times. Instead, you can call it once, get the value, then check that value and already have the `errorMessage` available if you needed. Then you can also just return that value instead of constructing the return object. This is similar to the change from the last iteration, and I would make the same change in the `checkOutputs` function below.

Reply:
sounds good